trace/trace.go

```diff
@@ -357,7 +357,7 @@ type Span interface {
 	// additional call to SetStatus is required if the Status of the Span should
 	// be set to Error, as this method does not change the Span status. If this
 	// span is not being recorded or err is nil then this method does nothing.
-	RecordError(err error, options ...EventOption)
+	RecordError(err error, options ...ErrorOption)
```

Comment:
I think changing this may warrant a major release.

Reply:
I think there must be an argument covariance/contravariance case I've missed here. I'll see if I can find a way of doing it and keeping full backwards compat.